src/lib/logger/index.js

```diff
@@ -21,6 +22,7 @@ const logger = createLogger({
 	defaultMeta: { service: 'tv-kitchen' },
 	transports: [
 		new transports.Console({
+			silent: isTestEnv() && !config.LOG_LEVEL,
```

Comment:
I'm a bit worried that this is hidden behavior being baked into `LOG_LEVEL` (as opposed to enabling something like `yarn test --verbose`), but I also don't want to belabor things too much and this is a non-critical worry that could easily be addressed if it becomes a priority down the line.

Reply:
Note: if we wanted to watch `--verbose` we could do something like

```
silent: isTestEnv() && process.argv.indexOf("--verbose") < 0,
```

Reply:
Note: I'm not playing passive disingenuously -- as I think through it I'm leaning to `--verbose` support instead of treating `LOG_LEVEL` as an override (the issue is `LOG_LEVEL` might be set for other reasons than wanting logs in tests.)

But I also appreciate that maybe `--verbose` is non-standard or otherwise bad.

Reply:
Jest has a `--verbose` flag. It "Display[s] individual test results with the test suite hierarchy."

Jest also has a `--silent` flag. It "Prevent[s] tests from printing messages through the console." That's the behavior we want, but it's not the default. We could change `yarn test` to do `jest --silent`, make this behavior parse out the silent flag on the command line (we probably need a new utility for that) and then override by just doing `jest`.

I'd argue using `LOG_LEVEL` makes some sense here. Basically I'm setting a `LOG_LEVEL` of silent (-1?) by default for tests, and then letting you choose a different one when you need it, the same way we have a default of info for other environments and let you override that.

Reply:
That makes sense to me!  This also starts to introduce the idea that `LOG_LEVEL` should generally be set on temporary basis for *either* use case.

Is `LOG_LEVEL` documented anywhere?  We should update the documentation to indicate that it will affect test output if set.

Other than that, this is good for merge I think!

Reply:
> Jest also has a `--silent` flag. It "Prevent[s] tests from printing messages through the console." That's the behavior we want, but it's not the default. We could change `yarn test` to do `jest --silent`, make this behavior parse out the silent flag on the command line (we probably need a new utility for that) and then override by just doing `jest`.

I'm not sure if this conflicts with the subsequent `LOG_LEVEL` discussion, but I do think having `jest --silent` disable the console transport is a great idea, and have our `yarn test` call `jest --silent` by default. (As noted we could call `jest` directly to get it non-silent, or we could have a `yarn test:noisy` variant if we wanted. I agree with Chris that we shouldn't overload `--verbose` here.)

Reply:
tbh I don't follow the `LOG_LEVEL` suggestions; I don't even know _how_ to set environment variables for the test environment, except as inline parameters in the yarn script definition.

Reply:
Following the `--silent` thread -- the reminder that we can, in fact, just call yarn jest in situations where we want to skip `--silent` is a really good point that I had not been considering, because I am a fool..

(side note we could also have a `yarn test:debug` if we really wanted to have that point documented, but meh).

Reply:
Doing some research on the behavior of silent. I'm curious about how it's supposed to work and why it doesn't with Winston. It does _something_ because as @slifty mentioned there's a difference between how the log output appears when you add the flag versus when you don't, but unlike `console.log` it doesn't succeed in suppressing it. More important, there's a `--no-silent` flag you can append after `--silent` to counteract it, like some sort of counter-spell that just solves the whole problem for us.

Reply:
Okay, so this is actually a weird interaction between Jest and Winston behavior:

Act 1: Jest normally hijacks the `console` with its own version called `BufferedConsole`, which pretends to be the `console` we know and love by providing its own writable stream instead of the standard output stream.

Act 2: Meanwhile, Winston's Console transport writes to `console._stdout` if it exists, `console.log` if it doesn't.

All good so far.

Act 3: But if we pass the `--silent` flag Jest instead subs in `NullConsole`, which is just a bunch of no-ops where logging functions should go. `stdout` doesn't get hijacked, so it just outputs.

* audience gasps *

Okay, so, what does this mean? It's possible we have the basis for Jest PR. But in the meantime, we could detect the `silent` flag by testing whether `console` is an instance of `NullConsole`. This is really testing specific, though, so it could go in a setup file.

Reply:
Excellent detective work, but just to check: we can't suss out `jest --silent` with `process.argv.indexOf("--silent") < 0` in a utility method? (I don't know if `process.argv` stays populated all the way through the… call stack? Am I using that word right??)

Reply:
I was going to reply with "yes of course we can but where's the fun in that," except, it appears no! We can't, because Jest runs the tests in a worker process that doesn't get the original args. Here's what I get in `argv` anyway:
```
/Users/chriszs/.nvm/versions/node/v14.4.0/bin/node,/Users/chriszs/Desktop/work/tv-kitchen/node_modules/jest-worker/build/workers/processChild.js
```
Thank god all that sleuthing had a purpose.

Reply:
Okay, just did as I suggested. See what you think. Assume we're not going to require JSDoc, unit tests for a test setup file, but can add those if needed.